Ignore empty YAML documents

I have been down a very deep rabbit hole today, and I want to share an experience I had with this tool (which is **wonderful**, thanks you very much!).

I was trying to generate a Terraform HCL document from a recent [linkerd](https://linkerd.io/2.10/getting-started/) installation. I was having a really hard time getting tfk8s to work at all, because it kept crashing like this:

```bash
[user@host ~/dev]$ cat linkerd.yaml | docker run --rm -i jrhouston/tfk8s:0.1.3 
panic: value is null

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.LengthInt(0x7b1d90, 0xa1dd60, 0x0, 0x0, 0xc000491c10)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1058 +0x28b
github.com/zclconf/go-cty/cty.Value.AsValueMap(0x7b1d90, 0xa1dd60, 0x0, 0x0, 0xc000308c00)
	/go/pkg/mod/github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1315 +0x9d
main.yamlToHCL(0x7b1d90, 0xa1dd60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x7b1d90, 0xa1dd60, ...)
	/build/tfk8s.go:87 +0x6a
main.YAMLToTerraformResources(0x7ad1a0, 0xc00000e010, 0x0, 0x0, 0x0, 0x0, 0x20, 0xc0000194ca, 0xc0001765a0)
	/build/tfk8s.go:157 +0x390
main.main()
	/build/tfk8s.go:199 +0x331
```

I found another very handy tool aptly named [kubernetes-split-yaml](https://github.com/mogensen/kubernetes-split-yaml). It also bailed, but it gave me a much better error message:

```
WARN[0000] Ignoring yaml file with kind missing         
```

So, I went digging into the `linkerd.yaml` file and there was an empty YAML document that was composed entirely of comments. I removed the empty object and `tfk8s` ran successfully after that. It's not a super hard thing for me to fix once I discovered what it was, but if there were a way to just ignore the empty docs or have an error message the entire experience would be improved.

--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 .PHONY: build docker docker-push release install test clean
 
-VERSION := 0.1.3
+VERSION := 0.1.4
 DOCKER_IMAGE_NAME := jrhouston/tfk8s
 
 build:

--- a/tfk8s.go
+++ b/tfk8s.go
@@ -154,6 +154,11 @@ func YAMLToTerraformResources(r io.Reader, providerAlias string, stripServerSide
 			return "", err
 		}
 
+		if doc.IsNull() {
+			// skip empty YAML docs
+			continue
+		}
+
 		formatted, err := yamlToHCL(doc, providerAlias, stripServerSide, mapOnly)
 
 		if err != nil {

--- a/tfk8s_test.go
+++ b/tfk8s_test.go
@@ -49,6 +49,11 @@ metadata:
 data:
   TEST: one
 ---
+# this empty
+# document 
+# should be 
+# skipped
+---
 apiVersion: v1
 kind: ConfigMap
 metadata:

